RES: Enable new name resolution engine by default

dima74 opened the pull request:
Enable [new name resolution engine](https://github.com/intellij-rust/intellij-rust/issues/6217) by default. There are still some corner cases unsupported, but I think that overall improvement from new resolve is big enough to enable it by default

Here is regressions tests comparison: [with build scripts evaulation](https://github.com/dima74/intellij-rust/actions/runs/745483458), [without build scripts](https://github.com/dima74/intellij-rust/actions/runs/745549334)

changelog: Enable new name resolution engine by default

Fixes #4093
Fixes #4811
Fixes #4887
Fixes #5007
Fixes #5146
Fixes #6236
Fixes #6243
Fixes #6270
Fixes #6302
Fixes #6355
Fixes #6452
Fixes #6486
Fixes #6863

Undin commented:
@dima74 Do I understand correctly, that all known issues listed in https://github.com/intellij-rust/intellij-rust/issues/6217#issuecomment-720347002 ?

Undin commented:
@dima74 Could you add all open issues to PR description if format `Fixes #issue_number` to close them when it will be merged, please?

dima74 commented:
> Do I understand correctly, that all known issues listed in [#6217 (comment)](https://github.com/intellij-rust/intellij-rust/issues/6217#issuecomment-720347002) ?

@Undin Yes, I updated https://github.com/intellij-rust/intellij-rust/issues/6217#issuecomment-720347002 with latest information

> Could you add all open issues related to [name resolution 2.0](https://github.com/intellij-rust/intellij-rust/labels/name%20resolution%202.0) to PR description in format `Fixes #issue_number` to close them when it will be merged, please?

Thanks, added

Undin reviewed (commented):
Let's also rename `Use experimental name resolution engine` in `Preferences | Languages & Frameworks | Rust` option to something like `Use new name resolution engine`. Otherwise, it will be quite weird that "experimental" option is enabled by default

dima74 commented:
bors r+
